app/models/user.rb

```diff
@@ -1,6 +1,17 @@
 class User < ApplicationRecord
+  GITHUB_USERNAME_PATTERN = /\A(?!\.)(?!_)[a-z0-9_\.]+(?<!\.)(?<!_)\z/
+
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+        :recoverable, :rememberable, :validatable
+
+  has_many :contacts, dependent: :destroy
+  has_many :friendships, dependent: :destroy
+  has_many :friends, through: :friendships
+  has_many :user_settings, dependent: :destroy
+  has_many :media, dependent: :destroy
+  has_many :reminders, dependent: :destroy
+
+  validates :username, uniqueness: true, format: { with: GITHUB_USERNAME_PATTERN, message: "Must be lowercase letters 🙇🏼‍♀️" }
```

Comment:
I don't speak regex but I think we need to update the message for this validation. Is not only validating that is lowercase no?

Reply:
Yes yes very true, did done ✅